docs/modelserving/autoscaling/keda/autoscaling_llm.md

```diff
@@ -0,0 +1,266 @@
+# Autoscale InferenceService with LLM metrics
+[KEDA (Kubernetes Event-driven Autoscaler)](https://keda.sh) is a lightweight, open-source component that extends Kubernetes' scaling capabilities by enabling event-driven scaling for any container workload, allowing applications to scale based on external metrics such as [vLLM metrics](https://docs.vllm.ai/en/latest/serving/metrics.html) for the number of waiting requests or KV Cache usage. 
+
+
+## Scale using the Metrics
+
+InferenceService scaling can be achieved in two ways:
+
+- **Using Metrics via Prometheus**: Scale based on Large Language Model (LLM) metrics collected in Prometheus.
+
+- **Using Metrics via OpenTelemetry**: Collect pod-level metrics (including LLM metrics) using OpenTelemetry, export them to the keda-otel-add-on gRPC endpoint, and use KEDA's external scaler for autoscaling.
+
+## Autoscale based on metrics from Prometheus
+
+Scale an InferenceService in Kubernetes using LLM (Large Language Model) metrics collected in Prometheus. 
+The setup leverages KServe with KEDA for autoscaling based on custom [Prometheus metrics](../../../modelserving/observability/prometheus_metrics.md).
+
+### Prerequisites
+
+1. Kubernetes cluster with KServe installed.
+
+2. [KEDA installed](https://keda.sh/docs/2.9/deploy/#install) for event-driven autoscaling.
+
+3. Prometheus configured to collect metrics from KServe.
+
+### Create `InferenceService`
+
+``` yaml
+kubectl apply -f - <<EOF
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: huggingface-fbopt
+  annotations:
+    serving.kserve.io/deploymentMode: "RawDeployment"
+    serving.kserve.io/autoscalerClass: "keda"
+    serving.kserve.io/enable-prometheus-scraping: "true"
+    prometheus.io/scrape: "true"
+    prometheus.io/path: "/metrics"
+    prometheus.io/port: "8080"
+    prometheus.io/scheme: "http"
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: huggingface
+      args:
+        - --model_name=fbopt
+        - --model_id=facebook/opt-125m
+      resources:
+        limits:
+          cpu: "1"
+          memory: 4Gi
+        requests:
+          cpu: "1"
+          memory: 4Gi
+    minReplicas: 1
+    maxReplicas: 5
+    autoScaling:
+      metrics:
+        - type: External
+          external:
+            metric:
+              backend: "prometheus"
+              serverAddress: "http://prometheus.istio-system.svc.cluster.local:9090"
```

Comment:
This will require the service mesh already installed and configured, right?
Maybe mention it in the prerequisites?